contrib/init/upstart/docker.conf

```diff
@@ -70,3 +70,25 @@ post-start script
 		echo "$DOCKER_SOCKET is up"
 	fi
 end script
+
+post-stop script
+        SHUTDOWN_RUNLEVELS='06'
+        if runlevel | cut -d' ' -f2 | grep -q [$SHUTDOWN_RUNLEVELS]; then
+                for container in $(docker-runc list -q); do
+                        docker-runc kill "$container" || true
```

Comment:
Yes, we should probably make `dockerd` aware of "why" it was stopped (so that it can either stop containers or keep them running). We now also have a custom `--stop-timeout` for containers (and allow them to be set _per container_ - especially for containers  that may need a longer time to gracefully shutdown.

ping @mlaventure @crosbymichael any thoughts?